### PR TITLE
Chapter 13: Micropost

### PR DESCRIPTION
**Pull request by rashidfirhat98**

### CODE REVIEW

**Comment by rashidfirhat98** [comment]
ready

File: app/controllers/microposts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  end
+
+  def destroy
+    @micropost.destroy
+    flash[:success] = "Micropost deleted"
+    redirect_to request.referrer || root_url
+  end
+
+  private
+  def micropost_params
```
Comment (saobv-2223) [inline app/controllers/microposts_controller.rb]:
```suggestion
  private
  
  def micropost_params
```

File: app/helpers/microposts_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module MicropostsHelper
+end
```
Comment (saobv-2223) [inline app/helpers/microposts_helper.rb]: remove file that never used

File: app/models/micropost.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+class Micropost < ApplicationRecord
+  belongs_to :user
+  scope :by_created_at, -> { order(created_at: :desc) }
+  has_one_attached :image
```
Comment (saobv-2223) [inline app/models/micropost.rb]:
```suggestion
  belongs_to :user
   has_one_attached :image
   
  scope :by_created_at, -> { order created_at: :desc }
 
```

File: app/models/micropost.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  validates :content, presence: true, 
+  length: {maximum: Settings.validate.post.max_length}
+
+  validate :image, content_type: { in: %w[image/jpeg image/gif image/png],
+                                  message: "must be a valid image format" },
+                           size: { less_than: 5.megabytes,
+                                  message: "should be less than 5MB" }
+  def display_image
+    image.variant resize_to_limit: [500, 500]
+  end
```
Comment (saobv-2223) [inline app/models/micropost.rb:19]:
```suggestion

  def display_image
    image.variant resize_to_limit: [500, 500]
  end       
```

File: app/views/shared/_micropost_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+<% end %>
+
+<script type="text/javascript">
+  $('#micropost_image').bind('change', function() {
+    var size_in_megabytes = this.files[0].size/1024/1024;
+    if (size_in_megabytes > 5) {
+      alert('Maximum file size is 5MB. Please choose a smaller file.');
+    }
+  });
+</script>
```
Comment (saobv-2223) [inline app/views/shared/_micropost_form.html.erb]: You should write js code in another file and call it here.
Do it if you can

File: app/views/static_pages/home.html.erb [hunk 6]
```diff
@@ -13,3 +13,22 @@
...
+        <%= render "shared/micropost_form" %>
+      </section>
+    </aside>
+    <div class="col-md-8">
+      <h3>Micropost Feed</h3>
+      <%= render "shared/feed" %>
+    </div>
+  </div>
+<% else %>
+
```
Comment (saobv-2223) [inline app/views/static_pages/home.html.erb]:
```suggestion
```

File: app/views/users/show.html.erb [hunk 7]
```diff
@@ -9,4 +9,15 @@
...
+      <h3>Microposts (<%= @user.microposts.count %>)</h3>
+      <ol class="microposts">
+        <%= render @microposts %>
+      </ol>
+      <%= paginate @microposts %>
+    <% end %>
+  </div>
 </div>
+
+
```
Comment (saobv-2223) [inline app/views/users/show.html.erb]:
```suggestion
```

**Comment by rashidfirhat98** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**